dev/PushNotifications/PushNotificationUtility.h

```diff
@@ -127,9 +130,18 @@ namespace winrt::Microsoft::Windows::PushNotifications::Helpers
     }
     CATCH_RETURN()
 
+    inline void ClearRoActivateInstanceCache()
+    {
+        // Activating a nonexistent class clears the cache enables the proxyStub to be found for elevated scenarios
+        auto activatableClass{ wil::make_unique_string_nothrow<wil::unique_hstring>(L"FakeClass.ClassName") };
+        winrt::com_ptr<::IInspectable> ptr{};
+
+        RoActivateInstance(activatableClass.get(), ptr.put());
+    }
+
     inline wil::com_ptr<INotificationsLongRunningPlatform> GetNotificationPlatform()
     {
-        return wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER);
+        return wil::CoCreateInstance<INotificationsLongRunningPlatform>(CLSID_NotificationsLongRunningPlatform, (CLSCTX_LOCAL_SERVER | CLSCTX_ALLOW_LOWER_TRUST_REGISTRATION));
```

Comment:
It's worth noting that since CLSCTX_ALLOW_LOWER_TRUST_REGISTRATION was serviced downlevel, there's a chance that passing the flag will cause CoCreateInstance to fail (even if the app is non-elevated) due to the flag being unrecognized on an unpatched machine. We did add a mechanism for detecting whether the flag is supported. So if you want to be robust against the possibility of this code running on unpatched downlevel machines, you might need to query.

`
// If supported, allow use of class registrations from less trusted sources.
auto supportMarker = wil::CoCreateInstanceNoThrow<ISupportAllowLowerTrustActivation>(CLSID_ActivationCapabilities);
if (supportMarker)
{
    activatedObject = wil::CoCreateInstance(CLSID_SampleClass, CLSCTX_LOCAL_SERVER |
        CLSCTX_ALLOW_LOWER_TRUST_REGISTRATION);
}
else
{
    // Even if we can't pass CLSCTX_ALLOW_LOWER_TRUST_REGISTRATION, we still want to attempt the activation.
    // This might fail to find the class when run elevated on a downlevel SKU, but it will work otherwise.
    activatedObject = wil::CoCreateInstance(CLSID_SampleClass, CLSCTX_LOCAL_SERVER);
}
`